serverless-plugin/index.js

```diff
@@ -186,57 +213,61 @@ class ServerlessPlugin {
                         libraries[name] = true;
                     }
                     foundLibraries[name] = true;
                 });
             }
-        });
-
-        // Error if a requested extra library could not be found
-        Object.keys(libraries).forEach(name => {
-            if (!libraries[name]) {
-                const msg = `Extra library not found: ${name}.`;
-                this.serverless.cli.log(msg);
-                // Show the libraries found, in case the name was misspelled
-                const foundLibrariesStr = Object.keys(foundLibraries)
-                      .filter(l => !IGNORE_LIBRARIES.includes(l))
-                      .sort()
-                      .join(", ");
-                this.serverless.cli.log(
-                    `Dependent libraries found: ${foundLibrariesStr}`);
-                throw new Error(msg);
-            }
-        });
-
-        // Create a shim to start the executable and copy it to all the
-        // destination directories
-        const handlerTemplate = fs.readFileSync(TEMPLATE).toString('utf8');
-
-        Object.keys(handlerOptions).forEach(packageName => {
-            let handler = handlerTemplate + handlerOptions[packageName].map(
-                ([executableName, options]) => `exports['${executableName}'] = wrapper(${JSON.stringify(options)});`
-            ).join("\n") + "\n";
 
-            const handlerFileName = path.resolve(this.servicePath, `${packageName}.js`);
-            fs.writeFileSync(handlerFileName, handler);
-            this.additionalFiles.push(handlerFileName);
+            // Error if a requested extra library could not be found
+            Object.keys(libraries).forEach(name => {
+                if (!libraries[name]) {
+                    const msg = `Extra library not found: ${name}.`;
+                    this.serverless.cli.log(msg);
+                    // Show the libraries found, in case the name was misspelled
+                    const foundLibrariesStr = Object.keys(foundLibraries)
+                          .filter(l => !IGNORE_LIBRARIES.includes(l))
+                          .sort()
+                          .join(", ");
+                    this.serverless.cli.log(
+                        `Dependent libraries found: ${foundLibrariesStr}`);
+                    throw new Error(msg);
+                }
+            });
+
+            // Create a shim to start the executable and copy it to all the
+            // destination directories
+            const handlerTemplate = fs.readFileSync(TEMPLATE).toString('utf8');
+
+            Object.keys(handlerOptions).forEach(packageName => {
+                let handler = handlerTemplate + handlerOptions[packageName].map(
+                    ([executableName, options]) => `exports['${executableName}'] = wrapper(${JSON.stringify(options)});`
+                ).join("\n") + "\n";
+
+                const handlerFileName = this.buildHandlerFileName(directory, packageName);
```

Comment:
Why isn't this adding `this.servicePath` anymore?

Reply:
I've changed the definition of buildHandlerFileName to include `this.servicePath`.